Generated by the following ruby script

puts(<<HEADER)
// +build amd64,solaris

package syscall

const (
HEADER

syscall_h = File.readlines('/usr/include/sys/syscall.h')
calls = {}
syscall_h.each do |line|
  m = line.match(/#define\s(SYS_[^\s]*)\s+(\d+)/)
  if m
    calls[m[1].upcase] = m[2]
  end
end
width = calls.keys.max_by { |x| x.length }.length + 1
calls.each do |call, num|
  printf("\t%-#{width}s = %d\n", call, num)
end
puts ")"

--- golang-1.6.2/src/syscall/zsysnum_solaris_amd64.go.orig	2016-04-27 22:29:15.592868144 +0300
+++ golang-1.6.2/src/syscall/zsysnum_solaris_amd64.go	2016-04-27 22:29:34.483620796 +0300
@@ -1,13 +1,250 @@
-// Copyright 2014 The Go Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style
-// license that can be found in the LICENSE file.
-
 // +build amd64,solaris
 
 package syscall
 
-// TODO(aram): remove these before Go 1.3.
 const (
-	SYS_EXECVE = 59
-	SYS_FCNTL  = 62
+	SYS_SYSCALL              = 0
+	SYS_EXIT                 = 1
+	SYS_READ                 = 3
+	SYS_WRITE                = 4
+	SYS_OPEN                 = 5
+	SYS_CLOSE                = 6
+	SYS_LINKAT               = 7
+	SYS_LINK                 = 9
+	SYS_UNLINK               = 10
+	SYS_SYMLINKAT            = 11
+	SYS_CHDIR                = 12
+	SYS_TIME                 = 13
+	SYS_MKNOD                = 14
+	SYS_CHMOD                = 15
+	SYS_CHOWN                = 16
+	SYS_BRK                  = 17
+	SYS_STAT                 = 18
+	SYS_LSEEK                = 19
+	SYS_GETPID               = 20
+	SYS_MOUNT                = 21
+	SYS_READLINKAT           = 22
+	SYS_SETUID               = 23
+	SYS_GETUID               = 24
+	SYS_STIME                = 25
+	SYS_PCSAMPLE             = 26
+	SYS_ALARM                = 27
+	SYS_FSTAT                = 28
+	SYS_PAUSE                = 29
+	SYS_STTY                 = 31
+	SYS_GTTY                 = 32
+	SYS_ACCESS               = 33
+	SYS_NICE                 = 34
+	SYS_STATFS               = 35
+	SYS_SYNC                 = 36
+	SYS_KILL                 = 37
+	SYS_FSTATFS              = 38
+	SYS_PGRPSYS              = 39
+	SYS_UUCOPYSTR            = 40
+	SYS_PIPE                 = 42
+	SYS_TIMES                = 43
+	SYS_PROFIL               = 44
+	SYS_FACCESSAT            = 45
+	SYS_SETGID               = 46
+	SYS_GETGID               = 47
+	SYS_MKNODAT              = 48
+	SYS_MSGSYS               = 49
+	SYS_SYSI86               = 50
+	SYS_ACCT                 = 51
+	SYS_SHMSYS               = 52
+	SYS_SEMSYS               = 53
+	SYS_IOCTL                = 54
+	SYS_UADMIN               = 55
+	SYS_FCHOWNAT             = 56
+	SYS_UTSSYS               = 57
+	SYS_FDSYNC               = 58
+	SYS_EXECVE               = 59
+	SYS_UMASK                = 60
+	SYS_CHROOT               = 61
+	SYS_FCNTL                = 62
+	SYS_ULIMIT               = 63
+	SYS_RENAMEAT             = 64
+	SYS_UNLINKAT             = 65
+	SYS_FSTATAT              = 66
+	SYS_FSTATAT64            = 67
+	SYS_OPENAT               = 68
+	SYS_OPENAT64             = 69
+	SYS_TASKSYS              = 70
+	SYS_ACCTCTL              = 71
+	SYS_EXACCTSYS            = 72
+	SYS_GETPAGESIZES         = 73
+	SYS_RCTLSYS              = 74
+	SYS_SIDSYS               = 75
+	SYS_LWP_PARK             = 77
+	SYS_SENDFILEV            = 78
+	SYS_RMDIR                = 79
+	SYS_MKDIR                = 80
+	SYS_GETDENTS             = 81
+	SYS_PRIVSYS              = 82
+	SYS_UCREDSYS             = 83
+	SYS_SYSFS                = 84
+	SYS_GETMSG               = 85
+	SYS_PUTMSG               = 86
+	SYS_LSTAT                = 88
+	SYS_SYMLINK              = 89
+	SYS_READLINK             = 90
+	SYS_SETGROUPS            = 91
+	SYS_GETGROUPS            = 92
+	SYS_FCHMOD               = 93
+	SYS_FCHOWN               = 94
+	SYS_SIGPROCMASK          = 95
+	SYS_SIGSUSPEND           = 96
+	SYS_SIGALTSTACK          = 97
+	SYS_SIGACTION            = 98
+	SYS_SIGPENDING           = 99
+	SYS_CONTEXT              = 100
+	SYS_FCHMODAT             = 101
+	SYS_MKDIRAT              = 102
+	SYS_STATVFS              = 103
+	SYS_FSTATVFS             = 104
+	SYS_GETLOADAVG           = 105
+	SYS_NFSSYS               = 106
+	SYS_WAITID               = 107
+	SYS_SIGSENDSYS           = 108
+	SYS_HRTSYS               = 109
+	SYS_UTIMESYS             = 110
+	SYS_SIGRESEND            = 111
+	SYS_PRIOCNTLSYS          = 112
+	SYS_PATHCONF             = 113
+	SYS_MINCORE              = 114
+	SYS_MMAP                 = 115
+	SYS_MPROTECT             = 116
+	SYS_MUNMAP               = 117
+	SYS_FPATHCONF            = 118
+	SYS_VFORK                = 119
+	SYS_FCHDIR               = 120
+	SYS_READV                = 121
+	SYS_WRITEV               = 122
+	SYS_PREADV               = 123
+	SYS_PWRITEV              = 124
+	SYS_GETRANDOM            = 126
+	SYS_MMAPOBJ              = 127
+	SYS_SETRLIMIT            = 128
+	SYS_GETRLIMIT            = 129
+	SYS_LCHOWN               = 130
+	SYS_MEMCNTL              = 131
+	SYS_GETPMSG              = 132
+	SYS_PUTPMSG              = 133
+	SYS_RENAME               = 134
+	SYS_UNAME                = 135
+	SYS_SETEGID              = 136
+	SYS_SYSCONFIG            = 137
+	SYS_ADJTIME              = 138
+	SYS_SYSTEMINFO           = 139
+	SYS_SHAREFS              = 140
+	SYS_SETEUID              = 141
+	SYS_FORKSYS              = 142
+	SYS_SIGTIMEDWAIT         = 144
+	SYS_LWP_INFO             = 145
+	SYS_YIELD                = 146
+	SYS_LWP_SEMA_POST        = 148
+	SYS_LWP_SEMA_TRYWAIT     = 149
+	SYS_LWP_DETACH           = 150
+	SYS_CORECTL              = 151
+	SYS_MODCTL               = 152
+	SYS_FCHROOT              = 153
+	SYS_VHANGUP              = 155
+	SYS_GETTIMEOFDAY         = 156
+	SYS_GETITIMER            = 157
+	SYS_SETITIMER            = 158
+	SYS_LWP_CREATE           = 159
+	SYS_LWP_EXIT             = 160
+	SYS_LWP_SUSPEND          = 161
+	SYS_LWP_CONTINUE         = 162
+	SYS_LWP_KILL             = 163
+	SYS_LWP_SELF             = 164
+	SYS_LWP_SIGMASK          = 165
+	SYS_LWP_PRIVATE          = 166
+	SYS_LWP_WAIT             = 167
+	SYS_LWP_MUTEX_WAKEUP     = 168
+	SYS_LWP_COND_WAIT        = 170
+	SYS_LWP_COND_SIGNAL      = 171
+	SYS_LWP_COND_BROADCAST   = 172
+	SYS_PREAD                = 173
+	SYS_PWRITE               = 174
+	SYS_LLSEEK               = 175
+	SYS_INST_SYNC            = 176
+	SYS_BRAND                = 177
+	SYS_KAIO                 = 178
+	SYS_CPC                  = 179
+	SYS_LGRPSYS              = 180
+	SYS_RUSAGESYS            = 181
+	SYS_PORT                 = 182
+	SYS_POLLSYS              = 183
+	SYS_LABELSYS             = 184
+	SYS_ACL                  = 185
+	SYS_AUDITSYS             = 186
+	SYS_PROCESSOR_BIND       = 187
+	SYS_PROCESSOR_INFO       = 188
+	SYS_P_ONLINE             = 189
+	SYS_SIGQUEUE             = 190
+	SYS_CLOCK_GETTIME        = 191
+	SYS_CLOCK_SETTIME        = 192
+	SYS_CLOCK_GETRES         = 193
+	SYS_TIMER_CREATE         = 194
+	SYS_TIMER_DELETE         = 195
+	SYS_TIMER_SETTIME        = 196
+	SYS_TIMER_GETTIME        = 197
+	SYS_TIMER_GETOVERRUN     = 198
+	SYS_NANOSLEEP            = 199
+	SYS_FACL                 = 200
+	SYS_DOOR                 = 201
+	SYS_SETREUID             = 202
+	SYS_SETREGID             = 203
+	SYS_INSTALL_UTRAP        = 204
+	SYS_SIGNOTIFY            = 205
+	SYS_SCHEDCTL             = 206
+	SYS_PSET                 = 207
+	SYS_SPARC_UTRAP_INSTALL  = 208
+	SYS_RESOLVEPATH          = 209
+	SYS_LWP_MUTEX_TIMEDLOCK  = 210
+	SYS_LWP_SEMA_TIMEDWAIT   = 211
+	SYS_LWP_RWLOCK_SYS       = 212
+	SYS_GETDENTS64           = 213
+	SYS_MMAP64               = 214
+	SYS_STAT64               = 215
+	SYS_LSTAT64              = 216
+	SYS_FSTAT64              = 217
+	SYS_STATVFS64            = 218
+	SYS_FSTATVFS64           = 219
+	SYS_SETRLIMIT64          = 220
+	SYS_GETRLIMIT64          = 221
+	SYS_PREAD64              = 222
+	SYS_PWRITE64             = 223
+	SYS_OPEN64               = 225
+	SYS_RPCSYS               = 226
+	SYS_ZONE                 = 227
+	SYS_AUTOFSSYS            = 228
+	SYS_GETCWD               = 229
+	SYS_SO_SOCKET            = 230
+	SYS_SO_SOCKETPAIR        = 231
+	SYS_BIND                 = 232
+	SYS_LISTEN               = 233
+	SYS_ACCEPT               = 234
+	SYS_CONNECT              = 235
+	SYS_SHUTDOWN             = 236
+	SYS_RECV                 = 237
+	SYS_RECVFROM             = 238
+	SYS_RECVMSG              = 239
+	SYS_SEND                 = 240
+	SYS_SENDMSG              = 241
+	SYS_SENDTO               = 242
+	SYS_GETPEERNAME          = 243
+	SYS_GETSOCKNAME          = 244
+	SYS_GETSOCKOPT           = 245
+	SYS_SETSOCKOPT           = 246
+	SYS_SOCKCONFIG           = 247
+	SYS_NTP_GETTIME          = 248
+	SYS_NTP_ADJTIME          = 249
+	SYS_LWP_MUTEX_UNLOCK     = 250
+	SYS_LWP_MUTEX_TRYLOCK    = 251
+	SYS_LWP_MUTEX_REGISTER   = 252
+	SYS_CLADM                = 253
+	SYS_UUCOPY               = 254
+	SYS_UMOUNT2              = 255
 )
